columnflow/tasks/ml.py

```diff
@@ -64,12 +70,33 @@ def __init__(self, *args, **kwargs):
                 f"{self.__class__.__name__}",
             )
 
+    @property
+    def preparation_producer_inst(self):
+        if self._preparation_producer_inst is not law.no_value:
+            # producer has already been cached
+            return self._preparation_producer_inst
+
+        producer = self.ml_model_inst.preparation_producer(self.config_inst)
+        if producer:
+            self._preparation_producer_inst = ProducerMixin.get_producer_inst(producer, {"task": self})
+
+            # overwrite the sandbox when set
+            if self._preparation_producer_inst.sandbox:
+                self.sandbox = self._preparation_producer_inst.sandbox
+
+        return self._preparation_producer_inst
```

Comment:
Mh, this potentially returns `law.no_value`. I think in case `preparation_producer` returns None, the `self._preparation_producer_inst` should also return None, which is does not right now.